### ISSUE TITLE
Can't quit using cmd-q with only dialog

### ISSUE THREAD

**Issue by jsoref** [TigerVNC/tigervnc#569]
![image](https://user-images.githubusercontent.com/2119212/34570336-6bd9188e-f139-11e7-98b1-6cc558f3a463.png)

macOS 10.13.1
TigerVNC 1.8.0

I should be able to Quit using cmd-q (or the menu item).
I can't because I haven't connected yet.
Instead, I have to use a mouse to hit the red gem (cmd-w doesn't work either...)

**Comment by CendioOssman** [comment]
So it works once you are connected?

I don't have any macOS machines here right now, but I can have a look once I do.

**Comment by jsoref** [comment]
I'm not sure actually. I think it does....

**Comment by CendioOssman** [comment]
I see this with the official builds, but not my own builds.

@bphinz, I think you need to debug this. What version of FLTK are you using?

**Comment by bphinz** [comment]
I believe that it's 1.3.4 but I will double check tonight.  @CendioOssman, what version of OS X are your builds produced on?

**Comment by CendioOssman** [comment]
We have our own build environment using gcc 5 and OS X 10.6 SDK.

**Comment by bphinz** [comment]
I'm using the 10.6 SDK as well.  I'll poke around tonight and see if anything is amiss.

**Comment by CendioOssman** [comment]
We're using an older FLTK (basically 1.3.3 plus patches). Might be a later bug in FLTK...

**Comment by bphinz** [comment]
Definitely using 1.3.4.  Can you send me the patches you're applying?

**Comment by CendioOssman** [comment]
I had a look and all our patches are just backports of stuff that is included in 1.3.4.

I'll try to do a test with 1.3.4 here and see if I get the same issue.

**Comment by CendioOssman** [comment]
Yeah, I'm getting the same issue after upgrading to FLTK 1.3.4. So it seems to be a bug there.

**Comment by CendioOssman** [comment]
I don't really have time to poke around with this right now. Could you check with the fltk devs, @bphinz?

**Comment by bphinz** [comment]
Yes, I'll try to chase this down.  I did a cursory check and didn't see any reports or google hits, but I'll post a question on the fltk dev forum and/or file a bug ticket.

**Comment by bphinz** [comment]
I did find this in the 1.3.4rc3 changelog:

<pre>
CHANGES IN FLTK 1.3.4rc3				RELEASED: Oct 31 2016

	Bug fixes and other improvements

	- Fix string truncation in fluid (STR #2722 continued).
	- MacOS: fix a small problem where the Command-Quit menu hotkey
	  may not work if Fl_Native_File_Chooser is used before any
	  window is shown.
</pre>

But I haven't found any reference to the related changeset.  Will probably have to do a diff between 1.3.4rc2 and 1.3.4rc3 to see what changed and if there's any relation to what we're doing (obviously we're not using a native file chooser before displaying any other window).

**Comment by CendioOssman** [comment]
An luck pinpointing this?

**Comment by bphinz** [comment]
No, I've been diffing 1.33 vs 1.34.  Here is the commit mentioned in the
1.3.4rc3 changelog:

https://github.com/fltk/test-only/commit/c3a098b486cc387089f93a50cac7738769b0de41

Maybe we need to add an fl_open_display?  I haven't found what changed that
would make 1.3.4 require this when 1.3.3 didn't though...

**Comment by CendioOssman** [comment]
Maybe. Have you tried calling it early and see what happens?

And did you ask on the FLTK mailing list?
